source/common/network/listener_impl.cc

```diff
@@ -28,13 +28,19 @@ void ListenerImpl::listenCallback(evconnlistener*, evutil_socket_t fd, sockaddr*
       final_local_address = original_local_address;
     }
 
-    // A listener that has the use_original_dst flag set to true can still receive connections
-    // that are NOT redirected using iptables. If a connection was not redirected,
-    // the address and port returned by getOriginalDst() match the listener port.
-    // In this case the listener handles the connection directly and does not hand it off.
-    if (listener->socket_.localAddress()->ip()->port() != final_local_address->ip()->port()) {
+    /**
```

Comment:
Best to use // C++ comment style for method internal comments. I think the Javadoc style is only for stuff that Doxygen processes.